src/compiler/config/test/validate-output-dist.spec.ts

```diff
@@ -107,4 +109,90 @@ describe('validateDistOutputTarget', () => {
     const { config } = validateConfig(userConfig, mockLoadConfigInit());
     expect(config.outputTargets.some((o) => o.type === 'dist')).toBe(false);
   });
+
+  it('sets option to transform aliased import paths when enabled', () => {
+    const outputTarget: d.OutputTargetDist = {
+      type: 'dist',
+      dir: 'my-dist',
+      buildDir: 'my-build',
+      empty: false,
+      transformAliasedImportPathsInCollection: true,
+    };
+    userConfig.outputTargets = [outputTarget];
+    userConfig.buildDist = true;
+
+    const { config } = validateConfig(userConfig, mockLoadConfigInit());
+
+    expect(config.outputTargets).toEqual([
+      {
+        buildDir: path.join(rootDir, 'my-dist', 'my-build'),
+        collectionDir: path.join(rootDir, 'my-dist', 'collection'),
+        copy: [],
+        dir: path.join(rootDir, 'my-dist'),
+        empty: false,
+        esmLoaderPath: path.join(rootDir, 'my-dist', 'loader'),
+        type: 'dist',
+        polyfills: undefined,
+        typesDir: path.join(rootDir, 'my-dist', 'types'),
+        transformAliasedImportPathsInCollection: true,
+      },
+      {
+        esmDir: path.join(rootDir, 'my-dist', 'my-build', 'testing'),
+        empty: false,
+        isBrowserBuild: true,
+        legacyLoaderFile: path.join(rootDir, 'my-dist', 'my-build', 'testing.js'),
+        polyfills: true,
+        systemDir: undefined,
+        systemLoaderFile: undefined,
+        type: 'dist-lazy',
+      },
+      {
+        copyAssets: 'dist',
+        copy: [],
+        dir: path.join(rootDir, 'my-dist', 'my-build', 'testing'),
+        type: 'copy',
+      },
+      {
+        file: path.join(rootDir, 'my-dist', 'my-build', 'testing', 'testing.css'),
+        type: 'dist-global-styles',
+      },
+      {
+        dir: path.join(rootDir, 'my-dist'),
+        type: 'dist-types',
+        typesDir: path.join(rootDir, 'my-dist', 'types'),
+      },
+      {
+        collectionDir: path.join(rootDir, 'my-dist', 'collection'),
+        dir: path.join(rootDir, '/my-dist'),
+        empty: false,
+        transformAliasedImportPaths: true,
```

Comment:
here we're basically testing that this is passed through from the user-supplied `'dist'` output target, right? just wondering if it makes sense to narrow this assertion down to just test that, something like

```ts
let distCollectionOT = config.outputTargets.find(ot => ot.type === "dist-collection");
expect(distCollectionOT.transformAliasedImportPaths).toBe(true);
```

idk, might just be a bit more compact :)

Reply:
IDK if I agree with this - the reason for which being is that tests like this, while admittedly wide in their assertions, are our primary way of checking if a change we've made affects other output targets as well.  Since we call `validateConfig` instead of `validateSpecifcOutputTarget`, we get all the side effects.  I would agree that what I'm describing isn't what is often taught as the "ideal" way of unit testing, it's our best way of catching regressions of this type today.  

Taking a step back, I'm also a bit squirmy about adding filtering logic to our tests - in my experience it makes it a little too easy for tests to creep in. e.g. if we change `isOutputTargetDist`, we've just affected the outcome of our tests too.

My vote is that we revert the changes to `src/compiler/config/test/validate-output-dist.spec.ts` in https://github.com/ionic-team/stencil/pull/3559/commits/c94c48836a1b467f96f6a438cf8e739b7c7cef23. Thoughts?

Reply:
I think I can go either way. I agree maybe the filter logic should be changed to just check the target `type` if we keep it. As for filtering the targets themselves, maybe we do revert at least for now. I think you make a good point that some of these targets are so coupled and tangled together that we might be safer checking the entirety of them. If/when we start to decouple them, then we could probably feel better about only checking certain ones

Reply:
mmmmm @rwaskiewicz I think you are sharing wisdom here and probably right :)